libraries/provider_slave_grants.rb

```diff
@@ -0,0 +1,45 @@
+class Chef
+  class Provider
+    # Used to setup permissions for MySQL replication user on master server
+    class MysqlmSlaveGrants < Chef::Provider::LWRPBase
+      use_inline_resources if defined?(use_inline_resources)
+
+      def whyrun_supported?
+        true
+      end
+
+      def action_create
+        grant_slave_privs
```

Comment:
+1 on using the mysql gem and running the queries through that instead of templating / executing / deleting. Using ruby instead of shelling out is more likely to work across a wider variety of platforms / setups.